tests/functional/testReindex.js

```diff
@@ -37,33 +100,173 @@ describe('UtapiReindex', () => {
-                    })
-                    .then(done)
-                    .catch(done);
+                shouldAcquireLock(done);
             });
         });
 
         describe('lock is already acquired', () => {
             beforeEach(done => {
-                reindex._lock()
-                    .then(res => {
-                        assert.strictEqual(res, 'OK');
-                    })
-                    .then(done)
-                    .catch(done);
+                shouldAcquireLock(done);
             });
 
             it('should not acquire the lock key', done => {
-                reindex._lock()
-                    .then(res => {
-                        assert.strictEqual(res, null);
-                    })
-                    .then(done)
-                    .catch(done);
+                shouldNotAcquireLock(done);
             });
         });
     });
+
+    describe('::_unlock', () => {
+        beforeEach(done => {
+            reindex._connect(done);
+        });
+
+        describe('lock is not acquired', () => {
+            it('should not release lock key', done => {
+                shouldNotReleaseLock(done);
+            });
+        });
+
+        describe('lock is already acquired', () => {
+            beforeEach(done => {
+                shouldAcquireLock(done);
+            });
+
+            it('should not acquire the lock key', done => {
+                shouldReleaseLock(done);
+            });
+        });
+    });
+
+    describe('::_attemptLock', () => {
+        beforeEach(done => {
+            reindex._connect(done);
+        });
+
+        describe('lock is not acquired', () => {
+            it('should call the job', done => {
+                const job = () => {
+                    done();
+                };
+                reindex._attemptLock(job);
+            });
+        });
+
+        describe('lock is already acquired', () => {
+            beforeEach(done => {
+                shouldAcquireLock(done);
+            });
+
+            it('should not call the job', done => {
+                const job = () => {
+                    done(new Error('job called when lock was not acquired'));
+                };
+                reindex._attemptLock(job);
+                setTimeout(done, 200);
+            });
+        });
+    });
+
+    describe('::_attemptUnlock', () => {
+        beforeEach(done => {
+            reindex._connect(done);
+        });
+
+        describe('lock is already acquired', () => {
+            beforeEach(done => {
+                shouldAcquireLock(done);
+            });
+
+            it('should release the lock key', done => {
+                reindex._attemptUnlock(); // Lock should be released here.
+                setTimeout(() => shouldNotReleaseLock(done), 200);
+            });
+        });
+    });
+
+    describe('::_scheduleJob', () => {
+        function waitUntilLockHasValue({ value }, cb) {
+            let shouldWait;
```

Comment:
It seems the meaning of this variable is reversed as it is used, you probably meant `shouldLeave`

Reply:
Thanks! I had changed `async.doWhile` to `async.doUntil` and neglected the naming of this variable.

Reply:
Updated.